src/analysis/processing/qgsalgorithmlayoutatlastoimage.cpp

```diff
@@ -0,0 +1,250 @@
+  const auto supportedImageFormats { QImageWriter::supportedImageFormats() };
+  for ( const auto format : QImageWriter::supportedImageFormats() )
+  {
+    if ( format == QByteArray( "svg" ) )
+      continue;
+    imageFormats << QString( format );
+  }
+  std::unique_ptr< QgsProcessingParameterEnum > extensionParam = qgis::make_unique< QgsProcessingParameterEnum >( QStringLiteral( "EXTENSION" ), QObject::tr( "Image format" ), imageFormats, false, imageFormats.indexOf( QStringLiteral( "png" ) ) );
+  extensionParam->setFlags( extensionParam->flags() | QgsProcessingParameterDefinition::FlagAdvanced );
+  addParameter( extensionParam.release() );
+
+  std::unique_ptr< QgsProcessingParameterNumber > dpiParam = qgis::make_unique< QgsProcessingParameterNumber >( QStringLiteral( "DPI" ), QObject::tr( "DPI (leave blank for default layout DPI)" ), QgsProcessingParameterNumber::Double, QVariant(), true, 0 );
+  dpiParam->setFlags( dpiParam->flags() | QgsProcessingParameterDefinition::FlagAdvanced );
+  addParameter( dpiParam.release() );
+
+  std::unique_ptr< QgsProcessingParameterBoolean > appendGeorefParam = qgis::make_unique< QgsProcessingParameterBoolean >( QStringLiteral( "GEOREFERENCE" ), QObject::tr( "Generate world file" ), true );
+  appendGeorefParam->setFlags( appendGeorefParam->flags() | QgsProcessingParameterDefinition::FlagAdvanced );
+  addParameter( appendGeorefParam.release() );
+
+  std::unique_ptr< QgsProcessingParameterBoolean > exportRDFParam = qgis::make_unique< QgsProcessingParameterBoolean >( QStringLiteral( "INCLUDE_METADATA" ), QObject::tr( "Export RDF metadata (title, author, etc.)" ), true );
+  exportRDFParam->setFlags( exportRDFParam->flags() | QgsProcessingParameterDefinition::FlagAdvanced );
+  addParameter( exportRDFParam.release() );
+
+  std::unique_ptr< QgsProcessingParameterBoolean > antialias = qgis::make_unique< QgsProcessingParameterBoolean >( QStringLiteral( "ANTIALIAS" ), QObject::tr( "Enable antialiasing" ), true );
+  antialias->setFlags( antialias->flags() | QgsProcessingParameterDefinition::FlagAdvanced );
+  addParameter( antialias.release() );
+}
+
+QgsProcessingAlgorithm::Flags QgsLayoutAtlasToImageAlgorithm::flags() const
+{
+  return QgsProcessingAlgorithm::flags() | FlagNoThreading;
+}
+
+QgsLayoutAtlasToImageAlgorithm *QgsLayoutAtlasToImageAlgorithm::createInstance() const
+{
+  return new QgsLayoutAtlasToImageAlgorithm();
+}
+
+QVariantMap QgsLayoutAtlasToImageAlgorithm::processAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+  // this needs to be done in main thread, layouts are not thread safe
+  QgsPrintLayout *layout = parameterAsLayout( parameters, QStringLiteral( "LAYOUT" ), context );
+  if ( !layout )
+    throw QgsProcessingException( QObject::tr( "Cannot find layout with name \"%1\"" ).arg( parameters.value( QStringLiteral( "LAYOUT" ) ).toString() ) );
+
+  QString error;
+  QgsLayoutAtlas *atlas = layout->atlas();
```

Comment:
This is fine for now, but we should add a way to clone a layout to avoid this in future.

Reply:
Yeah; I also was thinking it could be useful to have a QgsPrintLayout::setAtlas( QgsLayoutAtlas *atlas ) function too in the future.